File: app/schemas/pim.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field


PimSourceTypeValue = Literal["ftp", "sftp", "http", "local"]


class PimSourceCreateRequest(BaseModel):
    name: str = Field(min_length=1, max_length=255)
    type: PimSourceTypeValue
    host: str | None = Field(default=None, max_length=255)
    port: int | None = None
    username: str | None = Field(default=None, max_length=255)
    password: str | None = Field(default=None, max_length=255)
    path: str | None = Field(default=None, max_length=1024)
    file_pattern: str | None = Field(default=None, max_length=255)
    schedule: str | None = Field(default="0 2 * * *", max_length=255)
    is_active: bool = True


class PimSourceRead(BaseModel):
    id: int
    name: str
    type: PimSourceTypeValue
    host: str | None
    port: int | None
    username: str | None
    password: str | None
    path: str | None
    file_pattern: str | None
    schedule: str | None
    is_active: bool
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class PimRunImportRequest(BaseModel):
    source_id: int


class PimImportRunRead(BaseModel):
    id: int
    source_id: int | None
    status: str
    started_at: datetime
    finished_at: datetime | None
    records_processed: int
    records_created: int
    records_updated: int
    error_log: str | None
    source: PimSourceRead | None = None

    model_config = ConfigDict(from_attributes=True)


class PimConnectionTestResponse(BaseModel):
    ok: bool

File: admin-ui/lib/types/pim.ts
export type PimSourceType = "ftp" | "sftp" | "http" | "local";

export interface PimSource {
  id: number;
  name: string;
  type: PimSourceType;
  host: string | null;
  port: number | null;
  username: string | null;
  password: string | null;
  path: string | null;
  file_pattern: string | null;
  schedule: string | null;
  is_active: boolean;
  created_at: string;
}

export interface PimSourcePayload {
  name: string;
  type: PimSourceType;
  host?: string | null;
  port?: number | null;
  username?: string | null;
  password?: string | null;
  path?: string | null;
  file_pattern?: string | null;
  schedule?: string | null;
  is_active?: boolean;
}

export interface PimImportRun {
  id: number;
  source_id: number | null;
  status: string;
  started_at: string;
  finished_at: string | null;
  records_processed: number;
  records_created: number;
  records_updated: number;
  error_log: string | null;
  source: PimSource | null;
}

export interface PimConnectionTestResponse {
  ok: boolean;
}

export interface HejcoIntegrationSetting {
  provider: string;
  ftp_host: string | null;
  ftp_username: string | null;
  ftp_password_masked: string | null;
  has_password: boolean;
  pictures_path: string;
  product_data_path: string;
  stock_path: string;
  sync_enabled: boolean;
  sync_hour: number;
  timeout_seconds: number;
  last_sync_at: string | null;
  last_sync_status: string | null;
  last_sync_message: string | null;
  last_imported_product_count: number | null;
  created_at?: string | null;
  updated_at?: string | null;
}

export interface HejcoIntegrationPayload {
  provider?: string;
  ftp_host: string;
  ftp_username?: string | null;
  ftp_password?: string | null;
  pictures_path: string;
  product_data_path: string;
  stock_path: string;
  sync_enabled: boolean;
  sync_hour: number;
  timeout_seconds: number;
}

export interface HejcoConnectionTestResult {
  ok: boolean;
  message: string;
}

export interface HejcoSyncResult {
  products_imported: number;
  products_updated: number;
  images_matched: number;
  stock_updated: number;
  variants_imported: number;
  variants_updated: number;
  message: string | null;
}
